test/autorest/lrogroup/lros_test.go

```diff
@@ -37,22 +39,23 @@ func httpClientWithCookieJar() azcore.Transport {
 	})
 }
 
-// func TestLROResumeWrongPoller(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	resp, err := op.Delete202NoRetry204(context.Background())
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller := resp.GetPoller()
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	_, err = op.ResumeProductPoller(rt)
-// 	if err == nil {
-// 		t.Fatal("expected an error but did not find receive one")
-// 	}
-// }
+func TestLROResumeWrongPoller(t *testing.T) {
+	// TODO discuss what sort of check we want with the new design to avoid resuming from the wrong poller type
```

Comment:
Is this comment still applicable?

Reply:
Yes, we need to improve this functionality with the new changes you can resume pollers of the same type even if they were used for different operations. I'll add an issue as well

Reply:
Here's the issue #402

Reply:
I see.  I think you can fix this by having the `method` info in the token be a combination of the receiver type and method name.  What do you think?

Reply:
Oops `method` is the wrong thing.  I think it would be `pollerType`.  You could change this to capture the `receiver.methodName` info.

Reply:
We can fix it in a separate PR if you prefer.

Reply:
Something like ProductPollerDelete?

Reply:
Yes I'll add it in my next PR, but is my previous comment something we could do? Would it be a sufficient check?

Reply:
More like `lroRetrysOperations.Delete202Retry200`.  That way we know for sure that the poller is associated with this API.

Reply:
Ok will do in the next PR 👍